src/plugins/intel_npu/src/compiler/src/driver_compiler_adapter.cpp

```diff
@@ -25,113 +28,23 @@ LevelZeroCompilerAdapter::LevelZeroCompilerAdapter() : _logger("LevelZeroCompile
                        std::hex,
                        uint64_t(result));
     }
-    uint32_t drivers = 0;
-    result = zeDriverGet(&drivers, nullptr);
 
-    if (ZE_RESULT_SUCCESS != result) {
-        OPENVINO_THROW("L0 zeDriverGet get count",
-                       " result: ",
-                       ze_result_to_string(result),
-                       ", code 0x",
-                       std::hex,
-                       uint64_t(result));
-    }
+    ov::SoPtr<intel_npu::IEngineBackend> soPtrBackend = npuBackends->getIEngineBackend();
+    std::shared_ptr<intel_npu::IEngineBackend> iEngineBackend = soPtrBackend._ptr; // Extract the raw pointer
+    std::shared_ptr<ZeroEngineBackend> zeroBackend = std::dynamic_pointer_cast<ZeroEngineBackend>(iEngineBackend);
```

Comment:
Shall throw exception if fail

Reply:
Added a try catch and check if fail in casting